### ISSUE TITLE
Raffinements à l’aide de l’expression régulière (Regular expression refinements)

### ISSUE THREAD

**Issue by JazzfanRS** [JazzfanRS/Radiosure-station-database#9]
Est-il nécessaire de lister l’URL de streaming deux fois ? (Is it necessary to list the streaming URL twice?)

En outre, est-il possible de supprimer les quatre derniers champs vides? (Remove last 4 blank fields?)
`\t - \t - \t -\t`

Utilisez « regex » [^?] *$ (expression régulière) dans les URL de streaming pour supprimer le texte supplémentaire

C’est un bon moyen de supprimer les chaînes de 1000 caractères dans les URL Tick Tock

Merci! @francois-neosurf @GHbasicuser Il y a des dizaines d’autres petits raffinements que je pourrais suggérer (There dozens more small refinements I could suggest)

**Comment by GHbasicuser** [comment]
Hello @JazzfanRS 

Thanks for the translation. 😁 But I already use Google translate when you write long message. 😜 ( Via Firefox's "To Google Translate" extension)

>Est-il nécessaire de lister l’URL de streaming deux fois ? (Is it necessary to list the streaming URL twice?)

I noticed this in the RB2RS file... it's not a problem for RadioSure but I think you should ask @francois-neosurf. 😐 

>En outre, est-il possible de supprimer les quatre derniers champs vides? (Remove last 4 blank fields?)
\t - \t - \t -\t

Why don't you do the test via a database of a few lines ? 😉 If RadioSure says nothing, it should be considered.

>Utilisez « regex » [^?] *$ (expression régulière) dans les URL de streaming pour supprimer le texte supplémentaire

Yes, this should be done on the Radio-Browser side.  (There are about 56 urls over 1000 characters, up to almost 16000 for the longest. About 430 urls are over 1000 characters.)

I believe that the majority of streaming URLs don't need those parameters that follow the "question mark". On the other hand, they remain usable.

After sorting the lists of RSi and RS thoroughly (grouping of URL columns, deletion of duplicates URL before assembling the 2 lists, discarding http and https,..), **I found that there are almost 11000 radio streams in common**. And this does not include any duplicates that may exist in some playlists. Its represent 1/3 of radios in common.

**Comment by JazzfanRS** [comment]
@francois-neosurf @GHbasicuser I discovered that translatioon is difficult with large paragraphs and messages. I will stop.
I have downloaded the 5-12 update and deleted everything after the first streaming URL and RS operates as it should.
 
I have learned this: RS only requires the first six tabbed entries for a station to show and play correctly. In the 'details' window we see the streaming URL and five optional alternates URL fields that we do not use anymore.

Only five or more characters in the URL fields are passed to the 'details' window. Anything less is ignored.

The variation of ways to compare the two lists had proven too difficult for my hardware. After the PC meltdown from last weekend I decided if a simple script could find missing stations then that should be done.

I am still going to post the first 1000 stations I discovered missing. This will be on Google Sheets. And my post on Reddit will declare what is expected from the volunteers. RBI  submission is a disorderly and unregulated process, but it is all we have right now, 

@francois-neosurf Please update me with your progress, so I can know what to focus on.

**Comment by JazzfanRS** [comment]
This is what I am learning to do with each update I download. I have a long way before I know how to script. So I use an extended macro  NP++.

Delete line containing `"name": "'undefined' -> 'AFN 360 Benelux'",` as a duplicate of AFN 360 Benelux, which I submitted as a new station to replace poor information of the original. 

Delete spaces at the beginning and end of a line. Spaces preceding a station name are merely a 'top of the list' tactic used by submitters. 

Sort lexicographically ascending (symbols first, then numbers, then A to Z, Asian pictographs last)
                               I do this to aid me in finding poorly assembled information.

I will be updating my macro to include removing second URL stream to the end of line (EOL)

I haven't figured out how to handle the genre listings.

**Comment by francois-neosurf** [comment]
Hello to all,

Do you prefer I answer in English or in French ?

- Est-il nécessaire de lister l’URL de streaming deux fois ?
Les fichiers rsd d'origine que j'avais en exemple mentionnaient plusieurs URLs. Comme si il y avait une URL de secours en mode basse qualité. Je ne pense pas que ce soit indispensable, mais si Radiosure est capable de gérer plusieurs URLs, autant profiter de cette fonctionnalité.

- En outre, est-il possible de supprimer les quatre derniers champs vides?
I have exported a new RSD file and that sounds fine.
Je viens d'exporter un nouveau fichier rsd. Ca ne semble pas poser de problème.

- Delete line containing "name": "'undefined' -> 'AFN 360 Benelux'", as a duplicate of AFN 360 Benelux, which I submitted as a new station to replace poor information of the original.
Done

I have also fixed my automatic downloader.

Is there anything else to do ?

Regards.

**Comment by JazzfanRS** [comment]
My right click 'translate to english' on a web page does a good job of it even if it takes a several seconds to finsh.
 I thought to translate what I wanted to write would help here. Google translate doesn't like more then 3 or 4 sentences at once anyway.

RadioSure does allow for more then one stream URL, however it appears your list just shows the same one twice. 

I learned something new. First URL is primary source, if a URL is different it shows as an alternate. If same it is ignored. 
I checked this on 'details' page. RadioSure only shows one URL stream if the two are identical. 

With possible problems for some to connect with a stream, I will make instructions on how they can edit the particular station list to include alternates they might find. 

I do not see making a .php of our stations list that they can submit additions or changes to as an option. It risks the same vandalism as RBI.

As always do what you like in your free time. And weekly updates are fine.
